civictechprojects/forms.py

```diff
@@ -56,3 +58,44 @@ def create_project(request):
             thumbnail_json = json.loads(project_thumbnail_location)
             thumbnail = ProjectFile.from_json(project, FileCategory.THUMBNAIL, thumbnail_json)
             thumbnail.save()
+
+    @staticmethod
+    def edit_project(request, project_id):
+        project = Project.objects.get(id=project_id)
+
+        if not request.user.username == project.project_creator.username:
+            raise PermissionDenied()
+
+        form = ProjectCreationForm(request.POST)
+        project.project_description = form.data.get('project_description')
+        project.project_location=form.data.get('project_location')
+        project.project_name=form.data.get('project_name')
+        project.project_url=form.data.get('project_url')
+        issue_areas = form.data.get('project_issue_area')
+        if issue_areas and len(issue_areas) != 0:
+            Tag.merge_tags_field(project.project_issue_area, issue_areas)
+        project.save()
+
+        links_json_text = form.data.get('project_links')
+        if len(links_json_text) > 0:
+            links_json = json.loads(links_json_text)
+            ProjectLink.merge_changes(project, links_json)
+
+        files_json_text = form.data.get('project_files')
+        if len(files_json_text) > 0:
+            files_json = json.loads(files_json_text)
+            ProjectFile.merge_changes(project, files_json)
+
+        project_thumbnail_location = form.data.get('project_thumbnail_location')
+        if len(project_thumbnail_location) > 0:
+            thumbnail_json = json.loads(project_thumbnail_location)
+            existing_thumbnail = ProjectFile.objects\
```

Comment:
nit: I think you can avoid the backslash:

```
  existing_thumbnail = ProjectFile.objects.filter(
    file_project=project.id,
    file_category=FileCategory.THUMBNAIL.value,
  ).first()
```

Reply:
Done